dotcom-rendering/src/components/EnhancePinnedPost.importable.tsx

```diff
@@ -32,7 +19,7 @@ function toggleShowMore(show: boolean) {
 	if (pinnedPostButton) {
 		if (show) {
 			pinnedPostButton.style.removeProperty('display');
-		} else {
+		} else if (!checked) {
```

Comment:
This ensures that the “show less” button is shown, as the container is always high enough in that scenario.